CMakeLists.txt

```diff
@@ -298,16 +298,19 @@ endif()
 
 find_library(RE2 re2 REQUIRED)
 
-if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+set_source(ICU)
+if(CMAKE_SYSTEM_NAME MATCHES "Darwin" AND NOT (${ICU_SOURCE} STREQUAL "BUNDLED"
+                                              ))
   if(ON_APPLE_M1)
     set(CMAKE_PREFIX_PATH "/opt/homebrew/opt/icu4c" ${CMAKE_PREFIX_PATH})
   else()
     set(CMAKE_PREFIX_PATH "/usr/local/opt/icu4c" ${CMAKE_PREFIX_PATH})
   endif()
-  find_package(ICU REQUIRED)
-  include_directories(${ICU_INCLUDE_DIRS})
-  link_directories("${ICU_INCLUDE_DIRS}/../lib")
 endif()
+resolve_dependency(ICU)
+get_filename_component(ICU_LIBRARY_DIRS ${ICU_LIBRARIES} DIRECTORY)
+include_directories(${ICU_INCLUDE_DIRS})
+link_directories(${ICU_LIBRARY_DIRS})
```

Comment:
This behavior differs from the current status as ICU will now be added to the include/link directories on all platform and not only on macos. If this is not needed/detrimental for any reason we can easily restore the old behavior.

Reply:
@assignUser 
This broke the Presto build, it looks like.
https://github.com/prestodb/presto/pull/18816
https://github.com/prestodb/presto/actions/runs/3710302182/jobs/6289701898

```
-- Found glog: /usr/lib64/libglog.so  
-- Setting fmt source to AUTO
-- Setting ICU source to AUTO
CMake Error at velox/CMakeLists.txt:329 (get_filename_component):
  get_filename_component called with incorrect number of arguments
```